src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2577 @@
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopIncrement()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""); i = i.IndexOf(""x""$$))
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < s.IndexOf(""x""); i = i.IndexOf(""x"");$$)
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopBody()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < 10; i++)
+        {
+            i.ToString($$)
+        }
+";
+
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        string s = ""abcdefghij"";
+        for (int i = s.IndexOf(""bcd""); i < 10; i++)
+        {
+            i.ToString();$$
+        }
+";
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopObjectInitializer_MissingParen()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc""$$ }
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopObjectInitializer()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc""$$ } )
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+
+            VerifyNoSpecialSemicolonHandling(code);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ForLoopObjectInitializer_MissingBrace()
+        {
+            var code =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc""$$
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+            var expected =
+@"
+class C
+{
+    static void Main(string[] args)
+    {
+        for (Goo f = new Goo { i = 0, s = ""abc"";$$
+    }
+}
+public class Goo
+{
+    public int i;
+    public string s;
+}
+";
+            VerifyTypingSemicolon(code,expected);
+        }
+
+        #endregion
+
+        #region indexer
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void Indexer()
```

Comment:
>Indexer [](start = 20, length = 7)

Its confusing whether this test is trying to verify the bail out behavior for this feature within a body of an indexed accessor OR in context of an array element access. I am supposing it is the later, as it shouldn't really matter if you are within a regular method body, a property getter block body or indexed accessor block body. If so, consider changing this test to not have indexed property, but instead just a regular method body with array element access.

Reply:
Fixed